pkg/advisory/data_session.go

```diff
@@ -0,0 +1,240 @@
+	ds.workingBranch = branchName
+	wt, err := repo.Worktree()
+	if err != nil {
+		return nil, fmt.Errorf("getting worktree: %w", err)
+	}
+	err = wt.Checkout(&git.CheckoutOptions{
+		Branch: plumbing.NewBranchReferenceName(branchName),
+		Create: true,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("checking out new branch: %w", err)
+	}
+
+	// index advisory documents
+	index, err := v2.NewIndex(ctx, rwos.DirFS(tempDir))
+	if err != nil {
+		return nil, fmt.Errorf("indexing advisory documents: %w", err)
+	}
+	ds.index = index
+
+	return ds, nil
+}
+
+// Close closes the advisory data session and cleans up any temporary data that
+// was downloaded.
+func (ds DataSession) Close() error {
+	return os.RemoveAll(ds.tempDir)
+}
+
+// Create creates a new advisory within the context of the data session.
+func (ds *DataSession) Create(ctx context.Context, req Request) error {
+	err := Create(ctx, req, CreateOptions{
+		AdvisoryDocs: ds.index,
+	})
+	if err != nil {
+		return fmt.Errorf("creating advisory: %w", err)
+	}
+
+	err = ds.commit(ctx, req, "create")
+	if err != nil {
+		return fmt.Errorf("committing advisory creation: %w", err)
+	}
+
+	ds.modified = true
+	ds.modifiedPackages = append(ds.modifiedPackages, req.Package)
+
+	return nil
+}
+
+// Update updates an existing advisory within the context of the data session.
+func (ds *DataSession) Update(ctx context.Context, req Request) error {
+	err := Update(ctx, req, UpdateOptions{
+		AdvisoryDocs: ds.index,
+	})
+	if err != nil {
+		return fmt.Errorf("updating advisory: %w", err)
+	}
+
+	err = ds.commit(ctx, req, "update")
+	if err != nil {
+		return fmt.Errorf("committing advisory update: %w", err)
+	}
+
+	ds.modified = true
+	ds.modifiedPackages = append(ds.modifiedPackages, req.Package)
+
+	return nil
+}
+
+// Append creates a new event for an advisory if the advisory already exists, or
+// creates a new advisory with the event if the advisory does not already exist.
+func (ds *DataSession) Append(ctx context.Context, req Request) error {
+	packageSelection := ds.index.Select().WhereName(req.Package)
+	if packageSelection.Len() == 0 {
+		return ds.Create(ctx, req)
+	}
+
+	if _, exists := packageSelection.Configurations()[0].Advisories.GetByVulnerability(req.VulnerabilityID); !exists {
+		return ds.Create(ctx, req)
+	}
+
+	return ds.Update(ctx, req)
+}
+
+// Dir returns the path to the temporary directory where the session's advisory
+// data is currently stored.
+func (ds DataSession) Dir() string {
+	return ds.tempDir
+}
+
+// Index returns the index of advisory documents for the session.
+func (ds DataSession) Index() *configs.Index[v2.Document] {
+	return ds.index
+}
+
+// Modified returns true if any changes have been made to the advisory data
+// during the session.
+func (ds DataSession) Modified() bool {
+	return ds.modified
+}
+
+// Push pushes the changes made during the session to the remote advisories
+// repository.
+func (ds DataSession) Push(ctx context.Context) error {
```

Comment:
It doesn't seem we are using `Push` and OpenPullRequest anywhere else.

Reply:
What do you mean exactly?

https://github.com/luhring/wolfictl/blob/b553e73abef34930ebb96893c194e9f44af0a0b7/pkg/cli/advisory_guide.go#L352

https://github.com/luhring/wolfictl/blob/b553e73abef34930ebb96893c194e9f44af0a0b7/pkg/cli/advisory_guide.go#L357

Reply:
Okay, I see them now. It was github failed to me. 👍🏻